src/Core/Scheduler/Providers/WMS_Provider.js

```diff
@@ -45,53 +117,155 @@ WMS_Provider.prototype.preprocessDataLayer = function preprocessDataLayer(layer)
     layer.width = layer.heightMapWidth || 256;
     layer.version = layer.version || '1.3.0';
     layer.style = layer.style || '';
     layer.transparent = layer.transparent || false;
 
-    if (!layer.axisOrder) {
-        // 4326 (lat/long) axis order depends on the WMS version used
-        if (layer.projection == 'EPSG:4326') {
-            // EPSG 4326 x = lat, long = y
-            // version 1.1.0 long/lat while version 1.3.0 mandates xy (so lat,long)
-            layer.axisOrder = (layer.version === '1.1.0' ? 'wsen' : 'swne');
-        } else {
-            // xy,xy order
-            layer.axisOrder = 'wsen';
-        }
-    }
     let crsPropName = 'SRS';
     if (layer.version === '1.3.0') {
         crsPropName = 'CRS';
     }
 
-    layer.customUrl = `${layer.url
-                  }?SERVICE=WMS&REQUEST=GetMap&LAYERS=${layer.name
-                  }&VERSION=${layer.version
-                  }&STYLES=${layer.style
-                  }&FORMAT=${layer.format
-                  }&TRANSPARENT=${layer.transparent
-                  }&BBOX=%bbox` +
-                  `&${crsPropName}=${layer.projection
-                  }&WIDTH=${layer.width
-                  }&HEIGHT=${layer.width}`;
+    if (layer.extent && !(layer.extent instanceof Extent) && layer.projection) {
+        layer.extent = new Extent(layer.projection, layer.extent);
+    }
+
+    let getCapPromise;
+    if (layer.disableGetCap) {
+        getCapPromise = Promise.resolve();
+        if (!layer.projection) {
+            throw new Error('layer.projection is required');
+        }
+        if (!layer.extent && !layer.parentExtent) {
+            throw new Error('layer.extent is required');
+        }
+        layer.format = layer.format || 'image/png';
+    } else {
+        getCapPromise = Fetcher.xml(`${layer.url}?service=WMS&version=${layer.version}&request=GetCapabilities`, layer.networkOptions)
+            .then(xml => this.checkCapabilities(layer, xml));
+    }
+
+    return getCapPromise.then(() => {
+        if (!layer.axisOrder) {
+            // 4326 (lat/long) axis order depends on the WMS version used
+            if (layer.projection == 'EPSG:4326') {
+                // EPSG 4326 x = lat, long = y
+                // version 1.1.0 long/lat while version 1.3.0 mandates xy (so lat,long)
+                layer.axisOrder = (layer.version === '1.1.0' ? 'wsen' : 'swne');
+            } else {
+                // xy,xy order
+                layer.axisOrder = 'wsen';
+            }
+        }
+
+        layer.customUrl = `${layer.url
+        }?SERVICE=WMS&REQUEST=GetMap&LAYERS=${layer.name
+        }&VERSION=${layer.version
+        }&STYLES=${layer.style
+        }&FORMAT=${layer.format
+        }&TRANSPARENT=${layer.transparent
+        }&BBOX=%bbox` +
+        `&${crsPropName}=${layer.projection
+        }&WIDTH=${layer.width
+        }&HEIGHT=${layer.width}`;
+    });
+};
+
+WMS_Provider.prototype.findXmlLayer = function findXMLLayer(name, xml) {
```

Comment:
Could you add documentation on new functions?

Reply:
Oups I forgot! yes will do.

Reply:
Actually I did better: I only expose them through a `_testing` export, which avoid polluting the prototype of the provider with private functions.